content/rdi/architecture.md

```diff
@@ -0,0 +1,99 @@
+- Writing data - This is the part where Redis Data Integration writes the data to the target ensuring order & guarantying at least once to prevent data loss. In addition, the Writer can be configured to ignore, abort or keep rejected data entries.
+
+## Redis data integration control plane
+
+### RDI CLI
+
+Redis Data Integration comes with a python based CLI that is intuitive, self explanatory & validating to ease the entire lifecycle of Redis Data Integration setup and management.
+
+### RDI configuration
+
+Redis Data Integration configuration is persisted at the cluster level. The configuration is written by the CLI `deploy` command, reflecting the configuration file. This mechanism allows for automatic configuration of new shards when needed and can survive shard and node failure.
+
+### Secrets handling
+
+Redis Data Integration requires the following secrets:
+
+- The RDI Engine requires the credentials and certificate to connect and write encrypted data to the target Redis DB.
+- The Debezium Redis Sink Connector requires the credentials and certificate to connect and write data into Redis Data Integration DB.
+- The Debezium Source Connector requires the source DB secrets.
+- The RDI CLI requires a Certificate to connect to the Redis Enterprise cluster. For the create command it requires the credentials of a privileged user but it does not cache or store these credentials.
+
+Credentials and Certificate can be served in one of the following ways:
+
+- Entered manually as parameters to a CLI command (where applicable).
+- Stored in environment variables.
+- Injected to the temp file system by a secret store agent (e.g. [HashiCorp Vault](https://www.vaultproject.io/)).
+
+## Scalability and high availability
+
+Redis Data Integration is highly available:
+
+- At the feeder level: Debezium Server is deployed using [Kubernetes](https://kubernetes.io/) or [Pacemaker](https://clusterlabs.org/pacemaker/) for failover between stateless instances, while the state is secured in Redis.
+- At the Data and Control Plane: Using Redis Enterprise mechanisms for high availability (shard replica, cluster level configurations, etc.)
+
+Redis Data Integration is horizontally scalable:
+
+- Table level data streams can be split to multiple shards where they can be transformed and written to the target in parallel, while keeping table level order of updates.
+
+## Tested topologies
+
+Redis Data Integration is running on Redis Enterprise. It works on any installation of Redis Enterprise regardless of the runtime (bare metal, VMs or containers).
+Redis Data Integration can be collocated with the target Redis DB or run on a different Redis Enterprise cluster.
+
+Debezium Server should run in warm High Availability topology in one of the following options:
```

Comment:
I'm not sure about "run in...topology" here, but I'm also not entirely sure what wording would fit best here. Also, I'm also not sure what "warm" means in this context.
```suggestion
Debezium Server should run as a warm high availability topology in one of the following options:
```